assets/src/stories-editor/components/higher-order/with-right-click-handler.js

```diff
@@ -107,7 +135,7 @@ export default createHigherOrderComponent(
 			}
 
 			return (
-				<div onContextMenu={ onContextMenu }>
+				<div tabIndex="0" role="button" onContextMenu={ handleEvent } onKeyDown={ handleEvent }>
```

Comment:
Is button the best role here?

Reply:
The best I could find. It has to be an interactive role (otherwise linter complains about key handlers on non-interactive elements) and [the list of interactive roles](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-interactions.md#how-do-i-resolve-this-error) is very limited.

I'm open for other suggestions.

Reply:
Could we use the [`KeyboardShortcuts`](https://github.com/WordPress/gutenberg/tree/fec4a44ce4111bc3dfc628ae5ea4f58be1ebe6a5/packages/components/src/keyboard-shortcuts) components from the `@wordpress/components` package for this feature? It uses [Mousetrap](https://www.npmjs.com/package/mousetrap) under the hood. [Here is one usage example](https://github.com/WordPress/gutenberg/blob/fec4a44ce4111bc3dfc628ae5ea4f58be1ebe6a5/packages/block-editor/src/components/block-editor-keyboard-shortcuts/index.js) in Gutenberg.

It might then not be necessary to do any custom `role` stuff here.

Reply:
It's... tricky... but it might work.

Reply:
Fixed in 3fa7dd1. It actually wasn't that hard, but there is a pretty big limitation though. I don't know if it's an issue.

If you select an element with the mouse, you cannot press Shift+F10 (on Mac at least) to open the context menu (the focus is wrong and the `KeyboardShortcuts` handler doesn't trigger at all). However, if you tab to an element, then it works perfectly.

I think it's a bit unusual to use the mouse to navigate to the element but the keyboard to open the context menu, so I don't think it's an issue, but it might be. Will require real-world testing.

Reply:
Hmm in my testing it also only worked when tabbing, but only when the block's drag handle was selected (because it's a button for a11y reason).

I was not able at all to use it on the CTA block.